### PR TITLE
Update bwc test workflow to onboard BWC test on distribution level

### PR DESCRIPTION
**Pull request by zelinh**
Signed-off-by: Zelin Hao <zelinhao@amazon.com>

### Description
Update bwc test workflow to onboard BWC test on distribution level. 
Basically, as in our repo, we are targeting to run BWC on the distribution level. The command in our `bwctest.sh` script will be running command `./gradlew bwcTestSuite -Dtests.security.manager=false -PcustomDistributionDownloadType=bundle` with `customDistributionDownloadType` property set to `bundle`. In this way, the test cluster will be set up through the framework called within each plugin repo with the bundle distribution downloaded from our CI. 

### Issues Resolved
Part of #2870 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test_workflow/bwc_test/bwc_test_runner.py [hunk 1]
```diff
@@ -46,9 +46,9 @@ def run(self) -> TestSuiteResults:
                         test_results = test_suite.execute_tests()
                         all_results.append(component.name, test_results)
                     else:
-                        logging.info(f"Skipping bwc-tests for {component.name}, as it is currently not supported")
+                        logging.info(f"Skipping bwc-tests for {component.name}, as it is currently not supported according to the test manifest.")
```
Comment (peterzhuamazon) [inline src/test_workflow/bwc_test/bwc_test_runner.py:49]: What does `according to the test manifest` mean here.

**Review by peterzhuamazon — CHANGES_REQUESTED** [review]

File: src/test_workflow/bwc_test/bwc_test_runner.py (hunk 1 above)
Comment (peterzhuamazon) [reply src/test_workflow/bwc_test/bwc_test_runner.py:49]:
```
logging.info(f"Skipping bwc-tests for {component.name}, as the component in the test manifest has no bwc-tests section")
```

Comment (zelinh) [reply src/test_workflow/bwc_test/bwc_test_runner.py:49]: This log info means the component is declared in the manifest but it's not supported with bwc test. (e.g. ml-commons)
The one below means the component in our distribution manifest is not declared in the test manifest. e.g. (common-utils)

File: src/test_workflow/bwc_test/bwc_test_runner.py [hunk 2]
```diff
@@ -46,9 +46,9 @@ def run(self) -> TestSuiteResults:
...
                 else:
-                    logging.info(f"Skipping bwc-tests for {component.name}, as it is currently not declared in the test manifest")
+                    logging.info(f"Skipping bwc-tests for {component.name}, as it is currently not declared in the test manifest.")
```
Comment (peterzhuamazon) [inline src/test_workflow/bwc_test/bwc_test_runner.py:51]: Remove the dot.

**Review by peterzhuamazon — APPROVED** [review]

**Pull request merged**